frontend/src/components/GroupsFilterSelect.tsx

```diff
@@ -14,7 +14,9 @@ export default function GroupsFilterSelect({
   onChange,
 }: GroupsFilterSelectProps) {
   const options = Object.fromEntries(
-    groups.map((group) => [group.code, group.short_label || group.label]),
+    groups
+      .map((group) => [group.code, group.short_label || group.label])
+      .sort(),
```

Comment:
Same as above, it’s better to sort by the visible label (although in case of the current groups this is the same as sorting by code):

```suggestion
    groups
      .map((group) => [group.code, group.short_label || group.label])
      .sort((a, b) => a[1].localeCompare(b[1])),
```